docs/modules/modules-testdoc.adoc

```diff
@@ -0,0 +1,132 @@
+= TestDoc
+
+A TestDoc modul célja, hogy teszt osztályokból asciidoc dokumentációt generáljon.
+
+== Használat
+
+A TestDoc modul aktiválásához az alábbi függőséget kell meghivatkozni a pom.xml-ben:
+
+[source,xml]
+.pom.xml
+----
+<dependency>
+    <groupId>hu.icellmobilsoft.roaster</groupId>
+    <artifactId>roaster-testdoc</artifactId>
+</dependency>
+----
+
+Ezután a dokumentálni kívánt teszt osztályokat @TestDoc annotációval kell ellátni.
+
+Ha ezzel megvagyunk alapértelmezetten a fordításkor az elkészült `.jar` fájlba bekerül a generált asciidoc a `META-INF/test_doc.adoc` néven.
+Az egyes teszt osztályok külön táblázatonként jelennek meg a generált dokumentumban.
+
+== Példa
+
+*Input:*
+
+[source,java]
+----
+@TestDoc(titleHeadingLevel = 3) // <1>
+@DisplayName("Example test class")
+class TestDocExampleTest {
+
+    @Test // <2>
+    @DisplayName("Example test method in TestDocExampleTest class")
+    @Disabled("only for example")
+    void exampleTestMethod() {
+        Assertions.assertTrue(true);
+    }
+
+    @ParameterizedTest // <2>
+    @NullSource
+    @DisplayName("Example parameterized test method in TestDocExampleTest class")
+    @Disabled("only for example")
+    void exampleParameterizedTestMethod(String param) {
+        Assertions.assertNull(param);
+    }
+}
+
+
+
+
+@TestDoc(titleHeadingLevel = 3)  // <1>
+@DisplayName("Testing TestDoc functions")
+class TestDocTest {
+
+    @Test // <2>
+    @DisplayName("Generated file should contain the test data")
+    void generatedFileShouldContainTestData() throws URISyntaxException, IOException {
+        URL generatedFileUrl = getClass().getResource("/" + TestDocConfig.DEFAULT_OUTPUT_PATH + TestDocConfig.DEFAULT_OUTPUT_FILE_NAME);
+        assertNotNull(generatedFileUrl);
+
+        String generatedFile = Files.readString(Paths.get(generatedFileUrl.toURI()));
+
+        // classes
+        assertTrue(generatedFile.contains("Testing TestDoc functions"));
+        assertTrue(generatedFile.contains("Example test class"));
+
+        // methods
+        assertTrue(generatedFile.contains("generatedFileShouldContainTestData"));
+        assertTrue(generatedFile.contains("exampleTestMethod"));
+
+        // display names
+        assertTrue(generatedFile.contains("Generated file should contain the test data"));
+        assertTrue(generatedFile.contains("Example test method in TestDocExampleTest class"));
+    }
+}
+----
+
+<1> A generálást egy osztályon a `@TestDoc` annotációval aktiváljuk.
+A `titleHeadingLevel` paraméterrel állíthatjuk a generált táblázat címsorának a szintjét(opcionális).
```

Comment:
legyen itt pls, hogy mi a default, ha nincs megadva

Reply:
meg hogy milyen range-ben valid az értéke

Reply:
Ok, jó is hogy mondod, mert a range check logika lemaradt, azt is pótlom :)